pkg/resource-handler/controller/shard/shard_controller.go

```diff
@@ -189,33 +201,53 @@ func (r *ShardReconciler) reconcileMultiOrchService(
 }
 
 // reconcilePool creates or updates the StatefulSet and headless Service for a pool.
+// For pools spanning multiple cells, this creates one StatefulSet per cell.
 func (r *ShardReconciler) reconcilePool(
 	ctx context.Context,
 	shard *multigresv1alpha1.Shard,
 	poolName string,
-	poolSpec multigresv1alpha1.ShardPoolSpec,
+	poolSpec multigresv1alpha1.PoolSpec,
 ) error {
-	// Reconcile pool StatefulSet
-	if err := r.reconcilePoolStatefulSet(ctx, shard, poolName, poolSpec); err != nil {
-		return fmt.Errorf("failed to reconcile pool StatefulSet: %w", err)
+	// Pools must have cells specified
+	if len(poolSpec.Cells) == 0 {
+		return fmt.Errorf(
+			"pool %s has no cells specified - cannot deploy without cell information",
+			poolName,
+		)
```

Comment:
Similar to how MultiOrch was set to fail, Pool setup also requires the explicit list of cells right now. Should we correct this?